tools/cellprofiler/macros.xml

```diff
@@ -1,7 +1,11 @@
 <macros>
     <token name="@CP_VERSION@">3.1.9</token>
     <token name="@PY_VERSION@">2.7.16</token>
- 
+    <token name="@FORMATS@">jpg,png,tif,tiff,bmp,gif,pcx,ppm,psd,pbm,pgm,eps</token>
```

Comment:
is tif really a format? is it not just tiff?

Reply:
I think that the format is TIFF but the extensions can be .tif and .tiff. What's this `@FORMATS@` used for? filter file extensions, left?

Reply:
These should be strictly only Galaxy formats. Yi is controlling the file extension, and this is always what Galaxy allows. So I think the `tif` should be removed.

Reply:
Left, `tif` is not [here](https://github.com/galaxyproject/galaxy/blob/66db1c5c62c6ad12d5ffea6f312b50a44dac95fb/lib/galaxy/config/sample/datatypes_conf.xml.sample#L932), only `tiff`.

Reply:
i see galaxy has both tif and tiff listed as valid data types, and the format attributes uses file extension.
they are certainly no difference between these two,but I guess tif was there because some older file systems do not allow 4-letter extension. 
I will test it out and see what will happen.

Reply:
but it's `tif` as extension, not data type, left?

Reply:
[Here](https://github.com/galaxyproject/galaxy/blob/66db1c5c62c6ad12d5ffea6f312b50a44dac95fb/lib/galaxy/config/sample/datatypes_conf.xml.sample#L201):

    <datatype extension="tif" type="galaxy.datatypes.images:Tiff" subclass="true" display_in_upload="false"/>

Reply:
Mh, this is confusing. Not sure why we have two datatypes ... I would stick with `tiff` solely